ts/protocol/otf_message_handler.py

```diff
@@ -310,10 +312,26 @@ def _retrieve_input_data(conn):
 
     length = _retrieve_int(conn)
     value = _retrieve_buffer(conn, length)
-    if content_type == "application/json" and (decode_req is None or decode_req == "true"):
-        model_input["value"] = json.loads(value.decode("utf-8"))
-    elif content_type.startswith("text") and (decode_req is None or decode_req == "true"):
-        model_input["value"] = value.decode("utf-8")
+    if content_type == "application/json" and (
+        decode_req is None or decode_req == "true"
+    ):
+        try:
+            model_input["value"] = json.loads(value.decode("utf-8"))
+            return model_input
+
+        except Exception as e:
+            logging.exception(e)
+    elif content_type.startswith("text") and (
+        decode_req is None or decode_req == "true"
+    ):
+        try:
+            model_input["value"] = value.decode("utf-8")
+            return model_input
+
+        except Exception as e:
+            logging.exception(e)
+
     else:
         model_input["value"] = value
-    return model_input
+        return model_input
+    return None
```

Comment:
Not sure if returning `None` is the right approach here. The calling method `_retrieve_request` expects a `None` response when all the data from the socket has been read. In this case we would just end up sending an empty array to the handler which can be confusing for customers who did send an invalid json.
I think we should throw an exception here and have it handled up the stack so that a proper `400 Bad Request` error response can be created and returned to the customer

Reply:
I feel like 2 is probably the easiest. What it'll do is make the handler crash instead of the worker which feels more appropriate

Reply:
@maaquib just made the change, is thatt what you meant?

Reply:
@msaroufim  Added a couple of suggestions to make the log more direct

Reply:
hmm looks like everything broke, will take a look tmrw